chia/data_layer/dl_wallet_store.py

```diff
@@ -0,0 +1,329 @@
+import dataclasses
+from typing import List, Optional, Type, TypeVar, Union
+
+from aiosqlite import Row
+
+from chia.data_layer.data_layer_wallet import Mirror, SingletonRecord
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.db_wrapper import DBWrapper2
+from chia.util.ints import uint16, uint32, uint64
+from chia.wallet.lineage_proof import LineageProof
+
+_T_DataLayerStore = TypeVar("_T_DataLayerStore", bound="DataLayerStore")
+
+
+def _row_to_singleton_record(row: Row) -> SingletonRecord:
+    return SingletonRecord(
+        bytes32(row[0]),
+        bytes32(row[1]),
+        bytes32(row[2]),
+        bytes32(row[3]),
+        bool(row[4]),
+        uint32(row[5]),
+        LineageProof.from_bytes(row[6]),
+        uint32(row[7]),
+        uint64(row[8]),
+    )
+
+
+def _row_to_mirror(row: Row) -> Mirror:
+    urls: List[bytes] = []
+    byte_list: bytes = row[3]
+    while byte_list != b"":
+        length = uint16.from_bytes(byte_list[0:2])
+        url = byte_list[2 : length + 2]
+        byte_list = byte_list[length + 2 :]
+        urls.append(url)
+    return Mirror(bytes32(row[0]), bytes32(row[1]), uint64.from_bytes(row[2]), urls, bool(row[4]))
+
+
+class DataLayerStore:
+    """
+    WalletUserStore keeps track of all user created wallets and necessary smart-contract data
+    """
+
+    db_wrapper: DBWrapper2
+
+    @classmethod
+    async def create(cls: Type[_T_DataLayerStore], db_wrapper: DBWrapper2) -> _T_DataLayerStore:
+        self = cls()
+
+        self.db_wrapper = db_wrapper
+
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            await conn.execute(
+                (
+                    "CREATE TABLE IF NOT EXISTS singleton_records("
```

Comment:
Using a version field in this schema might help later.

Reply:
@Quexington, ping on a specific comment from Adam.